orange/src/main/java/com/hartwig/hmftools/orange/algo/OrangeAlgo.java

```diff
@@ -658,25 +659,45 @@ private Map<PercentileType, Evaluation> evaluateCohortPercentiles(@NotNull Orang
     }
 
     @NotNull
-    private OrangePlots buildPlots(@NotNull OrangeConfig config) throws IOException
+    private OrangePlots buildPlots(@NotNull OrangeConfig config, LinxData linxData) throws IOException
     {
         LOGGER.info("Loading plots");
 
         plotManager.createPlotDirectory();
 
         String linxPlotDir = config.linxPlotDirectory();
         List<String> linxDriverPlots = Lists.newArrayList();
-        if(new File(linxPlotDir).exists())
+
+        if(linxPlotDir != null)
         {
-            for(String file : new File(linxPlotDir).list())
+            if(new File(linxPlotDir).exists())
             {
-                linxDriverPlots.add(plotManager.processPlotFile(linxPlotDir + File.separator + file));
+                for(String file : new File(linxPlotDir).list())
+                {
+                    linxDriverPlots.add(plotManager.processPlotFile(linxPlotDir + File.separator + file));
+                }
             }
+
+            Set<Integer> linxReportableClusters;
+            try
+            {
+                linxReportableClusters =
+                        LinxReportableClusters.findReportableClusters(linxData, config.linxSomaticDataDirectory(), config.tumorSampleId());
+            }
+            catch(IOException e)
+            {
+                throw new RuntimeException("Unable to determine linx reportable clusters", e);
+            }
+
             LOGGER.info(" Loaded {} linx plots from {}", linxDriverPlots.size(), linxPlotDir);
+            if(linxReportableClusters.size() != linxDriverPlots.size())
+            {
+                throw new RuntimeException(String.format("Expected %d Linx plots but found %d", linxReportableClusters.size(), linxDriverPlots.size()));
+            }
```

Comment:
Are you confident your logic is always correct? 

I'd also be fine with a WARN in some situation I think, but leave that decision up to you. 

E.g. if there are 2 reportable clusters but 3 plots, would we be fine with a WARN?